ophyd/ophydobj.py

```diff
@@ -102,6 +99,14 @@ def register(instance):
     return weak_set
 
 
+class OphydObjectLoggerAdapter(LoggerAdapter):
+    """
+    A LoggerAdapter for use by OphydObject.
+    """
+    def process(self, msg, kwargs):
```

Comment:
[I neglected to comment on this in the last review, sorry.]

I believe that [`Formatters.format`](https://github.com/python/cpython/blob/9af0e47b1705457bb6b327c197f2ec5737a1d8f6/Lib/logging/__init__.py#L651-L681) is the intended place to do this kind of string formatting. I imagine that the rationale is that richer consumers, such as Elastic which can consume and index structured data, might not want or need the structure to be flattened into one message string, so that work should only be done for the handlers that require flat strings.

See [example from previous work](https://github.com/caproto/caproto/blob/334679b45fef52e998a4cbb398d95b5708b82c5b/caproto/_log.py#L107-L117).

I have no problem punting `Formatter` work to a separate PR, but I don't think we should grow a `LoggerAdapter` subclass in the mean time for the sake of doing work that fits well in a `Formatter`.